sw/host/tests/chip/flash_ctrl/src/main.rs

```diff
@@ -0,0 +1,293 @@
+    #[arg(long, value_parser = humantime::parse_duration, default_value = "600s")]
+    timeout: Duration,
+
+    /// Path to the firmware's ELF file, for querying symbol addresses.
+    #[arg(value_name = "FIRMWARE_ELF")]
+    firmware_elf: PathBuf,
+
+    #[command(flatten)]
+    sram_program: SramProgramParams,
+
+}
+
+fn test_update_phase(
+    _opts: &Opts,
+    test_word_address: u32,
+    transport: &TransportWrapper,
+    value: u32,
+) -> Result<()> {
+    let uart = transport.uart("console")?;
+    uart.set_flow_control(true)?;
+    let _ = UartConsole::wait_for(&*uart, r"Starting [^\r\n]*", _opts.timeout)?;
+    let _ = uart.clear_rx_buffer();
+    MemWrite32Req::execute(&*uart, test_word_address, value)?;
+    Ok(())
+}
+
+fn test_end(opts: &Opts, end_test_address: u32, transport: &TransportWrapper) -> Result<()> {
+    let uart = transport.uart("console")?;
+    let end_test_value = MemRead32Req::execute(&*uart, end_test_address)?;
+    assert!(end_test_value == 0);
+    MemWrite32Req::execute(&*uart, end_test_address, /*value=*/ 1)?;
+    let _ = UartConsole::wait_for(&*uart, r"test_end[^\r\n]*", opts.timeout)?;
+    Ok(())
+}
+
+fn test_sram_load(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
+    //
+    // Connect to the RISC-V TAP
+    //
+    transport.pin_strapping("PINMUX_TAP_RISCV")?.apply()?;
+    transport.reset_target(opts.init.bootstrap.options.reset_delay, true)?;
+
+    log::info!("Connecting to RISC-V TAP");
+    let mut jtag = opts
+        .init
+        .jtag_params
+        .create(transport)?
+        .connect(JtagTap::RiscvTap)?;
+
+    log::info!("Halting core");
+    jtag.halt()?;
+
+    opts.sram_program
+        .load_and_execute(&mut *jtag, ExecutionMode::Jump)?;
+
+    jtag.disconnect()?;
+    Ok(())
+}
+
+fn test_rma_command(opts: &Opts, transport: &TransportWrapper) -> anyhow::Result<()> {
+    let rma_bootstrap_strapping = transport.pin_strapping("RMA_BOOTSTRAP")?;
+    let tap_strapping = transport.pin_strapping("PINMUX_TAP_LC")?;
+
+    // Need to reset with `RMA_BOOTSTRAP` set.
+    rma_bootstrap_strapping
+        .apply()
+        .context("failed to apply RMA_BOOTSTRAP strapping")?;
+
+    // We keep `PINMUX_TAP_LC` set so that resets due to transition don't
+    // disconnect us from the LC TAP.
+    tap_strapping
+        .apply()
+        .context("failed to apply PINMUX_TAP_LC strapping")?;
+
+    transport
+        .reset_target(opts.init.bootstrap.options.reset_delay, true)
+        .context("failed to reset target")?;
+
+    log::info!("Connecting to JTAG interface");
+
+    let mut jtag = opts
+        .init
+        .jtag_params
+        .create(transport)?
+        .connect(JtagTap::LcTap)
+        .context("failed to connect to JTAG")?;
+
+    // Wait for the lifecycle controller to enter the `READY` state from
+    // which it accepts transition commands.
+    lc_transition::wait_for_status(&mut *jtag, Duration::from_secs(3), LcCtrlStatus::READY)
```

Comment:
Can this whole sequence of code be replace by [`trigger_lc_transition` instead of re-implementing it](https://github.com/lowRISC/opentitan/blob/master/sw/host/opentitanlib/src/test_utils/lc_transition.rs#L160) ?

Reply:
Replaced it. It looks much cleaner. thx.